blueprint-integration/src/main/java/com/github/dynamicextensionsalfresco/blueprint/DynamicExtensionsApplicationContext.java

```diff
@@ -239,23 +240,27 @@ protected EntityResolver getHostEntityResolver() {
         return getService(EntityResolver.class, HOST_APPLICATION_ALFRESCO_FILTER);
     }
 
-    /**
-     * Use the deprecated PackageAdmin to get the list of exported packages.
-     */
-    protected String[] getBundleExportPackages() {
-        String exportPackageHeader = this.getBundle().getHeaders().get(Constants.EXPORT_PACKAGE);
-        if (StringUtils.hasText(exportPackageHeader)) {
-            PackageAdmin packageAdmin = getService(PackageAdmin.class);
-            ExportedPackage[] packages = packageAdmin.getExportedPackages(this.getBundle());
-
-            String[] pkgNames = new String[packages.length];
-            for (int index = 0; index != packages.length; index++) {
-                pkgNames[index] = packages[index].getName();
+    private String[] getAllPackagesInBundle() {
+        final Set<String> packagesInBundle = new HashSet<>();
+
+        BundleWiring bundleWiring = this.getBundle().adapt(BundleWiring.class);
+        if (bundleWiring != null) {
+            Collection<String> classes = bundleWiring.listResources("/", "*.class",
```

Comment:
Looks a tad more complicated than I expected it to be.

Did you find a reference somewhere online that uses the same approach ?

Reply:
Based my method on this:
https://github.com/tjwatson/osgi-jpms-layer/blob/master/osgi.jpms.layer/src/osgi/jpms/internal/layer/BundleWiringPrivates.java#L43